### ISSUE TITLE
Make it easier to set the log threshold with an environment variable.

### ISSUE THREAD

**Issue by sellorm** [daroczig/logger#73]
I'm a big fan of setting log thresholds using environment variables so that people don't have to edit my apps to change the level.

This is really useful in production environments where the people running my code often do not know R and should not be editing any of the scripts, but who may need to change the log threshold of the application prior to execution.

In order to get this to work with logger, I end up having to do something like this:

``` r
log_threshold_from_env_var <- function(){
  log_level_env_var <- Sys.getenv("LOG_LEVEL", "INFO")
  log_levels <- c("FATAL", "ERROR", "WARN", "SUCCESS", "INFO", "DEBUG", "TRACE")
  if (! (log_level_env_var %in% log_levels)){
    err_msg <- "The LOG_LEVEL environment variable must be either unset, or set to a valid log level" 
    stop(err_msg)
  }
  get(log_level_env_var)
}
```

Which I can use as follows:

``` r
library(logger)
log_threshold(log_threshold_from_env_var())
```

This still defaults to INFO if the env var is unset.

Please could something like this be added to logger to make setting the log threshold from an environment variable more straightforward.

If there's a better way of doing this already, it's not clear from the documentation, so then I'd change this to a documentation request.

**Comment by daroczig** [comment]
Thanks a lot for the suggestion, I like the idea and easy to implement -- but can you please confirm which approach are you actually looking for?

* Have a helper fn like `log_threshold_from_env_var` made available in the `logger` package?
* Update `log_threshold` to default to read from an env var automatically when no `level` provided?
* Update the pkg to try to read the log level from an env var at the time of loading?
* Something else :) 

Eg (1) is trivial and I can make that happen today, (3) is also doable, but need some time to figure out how to do that properly.

**Comment by sellorm** [comment]
Ha! Yes, sorry, I should have been clearer about that.

Any of those options would be great.

However, Option 1 is very explicit about what's happening and it also doesn't change any existing behaviours, so I'd perhaps prefer that in the short term.

The third option would be really nice, but it feels like a fairly significant change from the way the package currently works. This feels like more of a long-term goal.

**Comment by Tamaraa01x** [comment]
[]()```
> #- [ ] [``]()
```

**Comment by daroczig** [comment]
Sorry that it took so long to get back to this .. but just planning a CRAN release, so I thought I should quickly check if the following works for you: https://github.com/daroczig/logger/pull/145/files

I'd highly appreciate any feedback before the end of this week, when I'm planning to submit a new version to CRAN. No hard feelings if you cannot get a chance to take a look after that many years, though :)

**Comment by AlexAxthelm** [comment]
That looks great to me!

**Comment by daroczig** [comment]
Implemented at #145
